app/src/main/java/richard/chard/lu/android/areamapper/activities/AreaMapperActivity.java

```diff
@@ -790,7 +795,7 @@ public void onSnapshotReady(Bitmap snapshot) {
                 snapshot,
                 Bitmap.CompressFormat.PNG,
                 100,
-                IMAGE_FILE_FOLDER,
+                getApplicationContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES).getPath() + File.separator + IMAGE_FILE_FOLDER,
```

Comment:
>  be used in SaveImageAsyncTask to save the file as well

Sorry this should not be necessary, I misunderstood how it's working.

Reply:
Yes, we can use our custom application class to share the app context.

Reply:
there should not be any need, I earlier thought that we are getting the image folder at 2 different places but we are not. I think we should just abstract this into a function like `getImageFolder()` inside `AreaMapperActivity` itself for code clarity otherwise it looks good to me.

Reply:
Got it, makes sense.